CMakeLists.txt

```diff
@@ -346,7 +346,7 @@ endif()
 
 option(WITH_TSAN "build with TSAN" OFF)
 if(WITH_TSAN)
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=thread -pie")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=thread")
```

Comment:
I thought TSAN required PIE.

Reply:
> I thought TSAN required PIE; could you clarify the purpose of this change?

Oh, never mind, it's implied when `-fsanitize=thread` is passed.

Reply:
I got exception when compile with `-WITH_TSAN=1`:
```
$ mkdir build; cd build; cmake .. -WITH_TSAN=1; make
...
[ 60%] Linking CXX executable c_test
clang: error: argument unused during compilation: '-pie' [-Werror,-Wunused-command-line-argument]
make[2]: *** [c_test] Error 1
...
```
And I didn't find doc about this option, so I guess it's out of date.

Reply:
Seems clang-9 does have `-pie` option: https://releases.llvm.org/9.0.0/tools/clang/docs/ClangCommandLineReference.html#cmdoption-clang-pie
but removed from clang-11: https://clang.llvm.org/docs/UsersManual.html

And seems GCC does need it for TSAN: https://mropert.github.io/2018/02/02/pic_pie_sanitizers/
I'll add it back.